src/main/java/com/ning/billing/recurly/RecurlyClient.java

```diff
@@ -1764,35 +1764,30 @@ private FluentCaseInsensitiveStringsMap callRecurlyNoContent(final AsyncHttpClie
             }
 
             // Handle errors payload
-            if (response.getStatusCode() >= 300) {
+          int statusCode = response.getStatusCode();
+          if (statusCode >= 300) {
                 log.warn("Recurly error whilst calling: {}\n{}", response.getUri(), payload);
-                RecurlyAPIError recurlyError = new RecurlyAPIError();
 
-                if (response.getStatusCode() == 422) {
-                    final Errors errors;
+                // 422 is returned for transaction errors (see https://dev.recurly.com/page/transaction-errors)
+                if (statusCode == 422) {
+                    Errors errors = null;
                     try {
                         errors = xmlMapper.readValue(payload, Errors.class);
                     } catch (Exception e) {
-                        // 422 is returned for transaction errors (see https://recurly.readme.io/v2.0/page/transaction-errors)
-                        // as well as bad input payloads
                         log.debug("Unable to extract error", e);
-                        return null;
                     }
+                  if (errors == null || (errors.getTransactionError() == null && errors.getRecurlyErrors() == null)) {
+                    throw new RecurlyAPIException(createRecurlyAPIError(payload, statusCode));
+                  } else {
                     throw new TransactionErrorException(errors);
-                } else if (response.getStatusCode() == 401) {
+                  }
+                } else if (statusCode == 401) {
+                    RecurlyAPIError recurlyError = new RecurlyAPIError();
```

Comment:
so we know we will never get recurlyerror in the payload if the response is 401?

Reply:
oh right it is done in another method, as a different error class??